plugins/CustomDimensions/API.php

```diff
@@ -62,13 +63,25 @@ public function getCustomDimension($idDimension, $idSite, $period, $date, $segme
 
         if (!empty($idSubtable) && $dataTable->getRowsCount()) {
             $parentTable = Archive::createDataTableFromArchive($record, $idSite, $period, $date, $segment);
-            foreach ($parentTable->getRows() as $row) {
-                if ($row->getIdSubDataTable() == $idSubtable) {
+
+            $parentValue = null;
+            if ($parentTable instanceof Map) {
+                $row = $parentTable->getRowFromIdSubDataTable($idSubtable);
```

Comment:
Doesn't this method exist for all DataTable types? Guess we could always use it to get the correct row in all cases. So we wouldn't need the check if it's a map.

Reply:
Good point :+1:  
I've updated the change to just use that method for all returned object and not check the type.